renku/cli/dataset.py

```diff
@@ -157,6 +158,39 @@ def add(client, name, urls, nocopy, relative_to, target, force):
         raise BadParameter('Could not process {0}'.format(url))
 
 
+@dataset.command('ls-files')
+@click.option(
+    '--dataset', multiple=True, help='Filter files in specific dataset.'
+)
+@click.option(
+    '--authors',
+    help='Filter files which where authored by specific authors. '
+    'Multiple authors are specified by comma.'
+)
+@click.option(
+    '--include', multiple=True, help='Include files matching given pattern.'
+)
+@click.option(
+    '--exclude', multiple=True, help='Exclude files matching given pattern.'
+)
+@click.option(
+    '--format',
+    type=click.Choice(LS_FILES_FORMATS),
+    default='tabular',
+    help='Choose an output format.'
+)
+@pass_local_client(clean=False, commit=False)
+def ls_files(client, format, exclude, include, authors, dataset):
+    """List files in dataset."""
+    LS_FILES_FORMATS[format](
+        client,
+        dataset_name=dataset,
+        authors=authors,
+        include=include,
```

Comment:
please move the `include` and `exclude` logic out from the formatter.

Reply:
I agree. Was thinking into extracting that part so I can reuse it on #447. Can you point me where should I put that part of code?

Reply:
I was thinking about something simple likeL
```python
records = _filter(datasets, dataset_name, authors, include, exclude)
LS_FILES_FORMATS[format](client, records)
```